DocumentsFromSnapshotMigration/src/test/java/com/rfs/FullTest.java

```diff
@@ -308,6 +316,112 @@ private DocumentsRunner.CompletionStatus migrateDocumentsWithOneWorker(SourceRep
+            final List<String> INDEX_ALLOWLIST = List.of();
+            CreateSnapshot.run(
+                    c -> new FileSystemSnapshotCreator(SNAPSHOT_NAME, c, SearchClusterContainer.CLUSTER_SNAPSHOT_DIR),
+                    new OpenSearchClient(esSourceContainer.getUrl(), null),
+                    false);
+            var tempDirSnapshot = Files.createTempDirectory("opensearchMigrationReindexFromSnapshot_test_snapshot");
+            var tempDirLucene = Files.createTempDirectory("opensearchMigrationReindexFromSnapshot_test_lucene");
+
+            String targetAddress = osTargetContainer.getHttpHostAddress();
+
+            String[] args = {
+                "--snapshot-name", SNAPSHOT_NAME,
+                "--snapshot-local-dir", tempDirSnapshot.toString(),
+                "--lucene-dir", tempDirLucene.toString(),
+                "--target-host", targetAddress
+            };
+
+            try {
+                esSourceContainer.copySnapshotData(tempDirSnapshot.toString());
+
+                var targetClient = new OpenSearchClient(targetAddress, null);
+                var sourceRepo = new FileSystemRepo(tempDirSnapshot);
+                migrateMetadata(sourceRepo, targetClient, SNAPSHOT_NAME, INDEX_ALLOWLIST);
+
+                // Stop the target container if we don't want it to be available.  We've already cached the address it was
+                // using, so we can have reasonable confidence that nothing else will be using it and bork our test.
+                if (!targetAvailable) {
+                    osTargetContainer.stop();
+                }
+
+                String classpath = System.getProperty("java.class.path");
+                String javaHome = System.getProperty("java.home");
+                String javaExecutable = javaHome + File.separator + "bin" + File.separator + "java";
+
+                // Kick off the doc migration process
+                log.atInfo().setMessage("Running RfsMigrateDocuments with args: " + Arrays.toString(args)).log();
+                ProcessBuilder processBuilder = new ProcessBuilder(
+                        javaExecutable, "-cp", classpath, "com.rfs.RfsMigrateDocuments"
+                );
+                processBuilder.command().addAll(Arrays.asList(args));
+                processBuilder.redirectErrorStream(true);
+
+                Process process = processBuilder.start();
+                log.atInfo().setMessage("Process started with ID: " + Long.toString(process.toHandle().pid())).log();
+                
+                // Kill the process and fail if we have to wait too long
+                int timeoutSeconds = 90;
+                boolean finished = process.waitFor(timeoutSeconds, TimeUnit.SECONDS);
+                if (!finished) {
+                    // Print the process output
+                    StringBuilder output = new StringBuilder();
+                    try (BufferedReader reader = new BufferedReader(new InputStreamReader(process.getInputStream()))) {
```

Comment:
Each of the 'new' objects should be in its own line to do proper cleanup for the try-with-resources.
```
try (
  var processStream = process.getInputStream();
  var isr = new InputStreamReader(processStream);
  var reader = new BufferedReader(isr)
) ...
```